code/src/components/Main.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable no-nested-ternary */
+import React from 'react';
+import { useSelector } from 'react-redux';
+import GameBoard from 'components/GameBoard';
+import UsernameInput from 'components/UsernameInput';
+import styled from 'styled-components/macro';
+import { LoadingIndicator } from 'components/LoadingIndicator';
+import { OuterWrapper, InnerWrapper } from '../utils/GlobalStyle';
+
+const Main = () => {
+  const isLoading = useSelector((store) => store.ui.isLoading);
+  const username = useSelector((store) => store.game.username);
+
+  return (
+    <OuterWrapper>
+      {/* If isLoading is true it shows the loading screen.
+       If there is a Username the gameboard shows. If not, the start page shows */}
+      <InnerWrapper>
+        {isLoading ? (
+          <LoadingIndicator />
+        ) : username ? (
+          <GameBoard />
+        ) : (
+          <StartSection>
+            <BewareText>
```

Comment:
Really cool text for getting the labyrinth feel but for the accesibilty it is a little bit hard to read. But then again it depends on what the purpose is of the text.